src/unitxt/inference.py

```diff
@@ -15,11 +17,27 @@ def _infer(self, dataset):
         """Perform inference on the input dataset."""
         pass
 
+    def _infer_log_probs(self, dataset):
```

Comment:
make abstract method like _infer

Reply:
I create a LogProbInferenceEngine API , so there is no need for all classes to implement this method.